### ISSUE TITLE
TypeVar not narrowed down after if+return

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#1870]
**Describe the bug**
Variables with a `TypeVar` type are not narrowed down after `if variable is ...: return`.

**To Reproduce**
```py
from pathlib import Path
from typing import TypeVar

# either None or sub-class of Path
PathNone = TypeVar("PathNone", None, Path)


def test(path: PathNone) -> PathNone:
    if path is None:
        return path
    reveal_type(path)  # mypy: pathlib.Path*, pyright: PathNone@test
    # ...
    return path
```

**Expected behavior**
`path` should be narrowed down to `pathlib.Path*`.

version 1.1.140

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.141, which I just published. It will also be included in the next release of pylance.
